lesson_05/homework_05.py

```diff
@@ -1,39 +1,91 @@
 small_list = [3, 1, 4, 5, 2, 5, 3]
 big_list = [3, 5, -2, -1, -3, 0, 1, 4, 5, 2]
+
 # task 1. Знайдіть всі унікальні елементи в списку small_list
+uniq_v = set(small_list)
+print(uniq_v)  # {1, 2, 3, 4, 5}
 
 # task 2. Знайдіть середнє арифметичне всіх елементів у списку small_list
+average = int(sum(small_list) / len(small_list))
+print(f"The arithmetic mean of the numbers in the list: {average}")
 
 # task 3. Перевірте, чи є в списку big_list дублікати
+big_set = set(big_list)
+if len(big_set) == len(big_list):
+    print("There are no duplicates in list!")
+else:
+    print("Opps...There are duplicates in list!")
+
 
+base_dict = {'contry': 'Ukraine', 'continent': 'Europe', 'size': 123}
+add_dict = {"a": 1, "b": 2, "c": 2, "d": 3, 'size': 12}
 
-base_dict = {'contry':'Ukraine', 'continent': 'Europe', 'size': 123}
-add_dict = {"a":1, "b":2, "c":2, "d":3, 'size': 12}
 # task 4. Знайдіть ключ з максимальним значенням у словнику add_dict
+max_key = max(add_dict, key=add_dict.get)
+print(f"The key with biggest value is '{max_key}'.")
 
 # task 5. Створіть новий словник, в якому ключі та значення будуть
 # замінені місцями у заданому словнику
+new_dict = {}
+for key, value in add_dict.items():
+    new_dict[value] = key
+
+print(new_dict)
 
-# task 6. Об'єднайте два словника base_dict та add_dict  в новий словник sum_dict
+# task 6. Об'єднайте два словники base_dict та add_dict в новий словник sum_dict
 # Якщо ключі збігаються, то перетворіть значення в строку та об'єднайте їх
-sum_dict = {}
+sum_dict = base_dict.copy()
+
+for k, v in add_dict.items():
+    if k in sum_dict:
+        sum_dict[k] = f"{str(sum_dict[k])}, {str(v)}"
```

Comment:
Тут малося нна увазі, щось типу
`sum_dict[k] = str(sum_dict[k]) +str(v)`
а якщо як у вас, то  f-стрінга тим і гарна, що автоматом все перетворює в стріг, тобто треба так:
```suggestion
        sum_dict[k] = f"{sum_dict[k]}, {v}"
```

Reply:
поняла, спасибо!)